src/compiler/AST/index.js

```diff
@@ -166,7 +166,7 @@ export function numberLiteral(
   }
 
   return {
-    type: 'NumberLiteral',
+    type: type === 'i64' ? 'LongLiteral' : 'NumberLiteral',
```

Comment:
We actually discussed about that with @maurobringolf offline (next time we will make an issue, sorry).

I saw that `i8` and `i16` were also around (unspecified now tho) and proposed to introduce more granular types:

```js
// using flow syntax

interface NumberLiteral {
  type: 'NumberLiteral';
  value: any;
}

type i8 = NumberLiteral & { value: number; }
type i16 = NumberLiteral & { value: number; }
type i32 = NumberLiteral & { value: number; }
type i64 = NumberLiteral & { hi: i32; low: i32; } // because JS can't handle 64 bits int
```

And the according AST builders.

Do we agree on this? cc @maurobringolf

Reply:
Yes agree 👍